2021/test/UltimateGoalAuto.java

```diff
@@ -9,29 +9,42 @@
 
 import org.firstinspires.ftc.robotcore.external.Telemetry;
 
+//import opmodes.SkyStoneAutoTwoStone;
 import team25core.DeadReckonPath;
 import team25core.DeadReckonTask;
 import team25core.GamepadTask;
 import team25core.MechanumGearedDrivetrain;
+import team25core.RingDetectionTask;
 import team25core.Robot;
 import team25core.RobotEvent;
+import team25core.SingleShotTimerTask;
 import team25core.StandardFourMotorRobot;
+import team25core.StoneDetectionTask;
 
 
-@Autonomous(name = "Scrimmage2", group = "Team 25")
+@Autonomous(name = "Scrimmage4", group = "Team 25")
 // @Disabled
 public class UltimateGoalAuto extends Robot {
```

Comment:
This is not a test opmode.  It should not live in the test directory.

You should already have an opmode directory under 2021.  This should move there.

The test directory should be reserved exclusively for unit test opmodes.  By which I mean opmodes that serve only to test an isolated portion of the robot.  A single servo, a mechanism, an opmode that all it does it test ring detection, nothing else, which means you aren't running that opmode as an autonomous in competition.  All of your competition opmodes should live elsewhere.

Along those lines, I presume the SkyStone sample above is there because it was used as a reference for work for ring detection.  That's fine, but it should go in a samples directory, not the test directory.

Reply:
Hi Craig my latest commit has the fixes based on your feedback could you please review them.  I moved RingImageInfo to terkel and reorganized the code to opmodes, samples, and test.